AnkiDroid/src/main/java/com/ichi2/async/Connection.java

```
@@ -249,13 +251,16 @@ private Payload doInBackgroundLogin(Payload data) {
     }
 
 
-    private boolean timeoutOccured(Exception e) {
+    private boolean timeoutOccurred(Exception e) {
         String msg = e.getMessage();
         return msg.contains("UnknownHostException") ||
                 msg.contains("HttpHostConnectException") ||
                 msg.contains("SSLException while building HttpClient") ||
                 msg.contains("SocketTimeoutException") ||
                 msg.contains("ClientProtocolException") ||
+                msg.contains("deadline reached") ||
```

Probably with the new library we no longer need to rely on string comparisons of the message to figure out whether a network error occurred... This was just a nasty hack we threw together to work around that old Apache library. I'll merge for now as a quick fix, but you should definitely consider using they type information instead of string comparisons if possible.

---

I actually did consider it - 

okhttp pops a 'java.io.InterruptedIOException' which is a java.io.IOException subclass, or a 'java.net.ConnectException' so it's possible ...but all the exceptions are actually swallowed and re-thrown as RuntimeException, which bubble up to a nested try where some types are caught, and then finally to an outer try that catches everything, where the message is then inspected in a few ways, and if it's known then the return is one of a bunch of hard-coded strings as an AsyncTask data object return value

The whole thing was so convoluted I dared not touch it right now, even though I was tempted - I'd rather get closure on the stack of existing PRs :-)